### DISCUSSION TITLE
Extraction as a JsonPrimitive / Find type of value for a key ?

### DISCUSSION THREAD

**Discussion by Shabinder** [russhwolf/multiplatform-settings#207] [Q&A] [answered]
I have a use case of backing up all key/values from `Settings`, since all values are stored as primitives, exporting key/value pairs into a JsonObject made sense, since I have access to `Settings.keys`, I hoped to get value by just iterating over the primitives using `getXorNull(key)`.

It worked on Android, but not on Desktop, since getIntOrNull, always returns 0, if value of another type.

I found that code does mention that behavior is undefined in this case, so wanted to know if there is a better way to approach this, was trying not to get into expect/actual into platform specific impl, to get export working.

So, is there a better way to do this ?

**Comment by russhwolf** [comment, marked as answer]
There's not really a good built-in for this, sorry. I explicitly left that behavior undefined to avoid the complication of massaging every platform API to have the same behavior since there's so many little edge-cases there. 

If you know all possible keys and types explicitly, you can probably do something clever with the serialization extension to convert to json. But if you know that then you wouldn't need to iterate the way you are.

One option might be to store everything as strings, assuming you have the flexibility to change that.

**Reply by Shabinder** [reply]
I ended up doing that only, solved my use case, with negligible overhead.
